### PR TITLE
Added exception to give an extra control to developers

### PR DESCRIPTION
**Pull request by ggnet**

### CODE REVIEW

File: Exception/LifecircleException.php [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+
+namespace Ecentria\Libraries\EcentriaAPIEventsBundle\Exception;
+
+/**
+ * Abstract Flow Exception
+ * 
+ * @copyright   2016 OpticsPlanet, Inc
+ * @author      Eugene Boiarynov <ievgen.boiarynov@opticsplanet.com>
+ */
+abstract class LifecircleException extends \Exception
```
Comment (ktsharp) [inline Exception/LifecircleException.php]: Should be LifecycleException, which makes more grammatical sense

Comment (ktsharp) [inline Exception/LifecircleException.php]: It's Lifecycle, not Lifecircle.

Can you explain why you called the base class "LifecycleException"? I think ConsumerException would be a better name.

File: Exception/AckException.php [hunk 2]
```diff
@@ -0,0 +1,32 @@
+<?php
```
Comment (ktsharp) [inline Exception/AckException.php:1]: Seems odd to have an Exception for Acknowledging that the message was processed successfully.  I suppose there are situations when you'd want it but it seems odd...

**Review by ktsharp — CHANGES_REQUESTED** [review]
See 2 comments/questions

File: Consumer/MessageConsumer.php [hunk 3]
```diff
@@ -74,9 +75,14 @@ public function execute(AMQPMessage $msg)
...
-        $this->messageDispatcher->dispatchMessage($message);
+        try {
+            $message = $this->serializer
+                ->deserialize($msg->body, $this->messageClassName, 'json');
+            $this->messageDispatcher->dispatchMessage($message);
+        } catch (LifecircleException $e) {
+            return $e->getFlag();
+        }
+        
         return true;
```
Comment (baziak3) [inline Consumer/MessageConsumer.php:89]: Is it mistake in parent interface description or in this implementation?
`@return` description inside PHPDoc is very confusing. Even though it is copied from ConsumerInterface, and might be wrong there, it should be updated in our implementation with what's actually going on in the method.

**Pull request merged**